src/test/java/com/rtb/projectmanagementtool/project/ProjectDataTest.java

```diff
@@ -54,123 +58,145 @@ public void tearDown() {
 
   // Test project creation given name, desc, and creator
   @Test
-  public void testCreateProject() {
+  public void createProject() {
     ProjectData project1 = new ProjectData(PROJECT1_NAME, PROJECT1_DESC, USER2);
-    Assert.assertEquals(project1.getName(), PROJECT1_NAME);
-    Assert.assertEquals(project1.getDescription(), PROJECT1_DESC);
-
-    ArrayList<Long> expectedTasks = new ArrayList<Long>();
-    ArrayList<String> expectedUsers = new ArrayList<String>();
-    expectedUsers.add(USER2_CREATOR);
 
-    Assert.assertEquals(expectedTasks, project1.getTasks());
-    Assert.assertEquals(expectedUsers, project1.getUsers());
+    Assert.assertEquals("project name is correct", project1.getName(), PROJECT1_NAME);
+    Assert.assertTrue("project creator id is correct", project1.getCreatorId() == USER2);
+    Assert.assertEquals(
+        "project derscription is correct", project1.getDescription(), PROJECT1_DESC);
+    Assert.assertEquals("no tasks in project", Collections.emptySet(), project1.getTasks());
+    Assert.assertEquals(
+        "no admins in project",
+        Collections.emptySet(),
+        project1.getUsers().get(UserProjectRole.ADMIN));
+    Assert.assertEquals(
+        "no members in project",
+        Collections.emptySet(),
+        project1.getUsers().get(UserProjectRole.MEMBER));
   }
 
   // Test project creation from entity
   @Test
   public void createProjectFromEntity() {
     // Create entity
     Entity entity = new Entity("Project");
-    entity.setProperty("name", PROJECT1_NAME);
-    entity.setProperty("description", PROJECT1_DESC);
-    entity.setProperty("tasks", new ArrayList<Long>());
-    ArrayList expectedUsers = new ArrayList<String>(Arrays.asList(USER2_CREATOR));
-    entity.setProperty("users", expectedUsers);
+    entity.setProperty(PROPERTY_NAME, PROJECT1_NAME);
+    entity.setProperty(PROPERTY_CREATOR, USER2);
+    entity.setProperty(PROPERTY_DESCRIPTION, PROJECT1_DESC);
+    entity.setProperty(PROPERTY_TASKS, new HashSet<Long>());
+    entity.setProperty(PROPERTY_ADMINS, new HashSet<Long>(Arrays.asList(USER3, USER4)));
+    entity.setProperty(PROPERTY_MEMBERS, new HashSet<Long>(Arrays.asList(USER1)));
 
     // Create project
     ProjectData project = new ProjectData(entity);
-    Assert.assertEquals(expectedUsers, project.getUsers());
+
+    // Have to use toString() because while entities have the same variables, they don't have the
+    // same references
+    Assert.assertEquals("project matches entity", entity.toString(), project.toEntity().toString());
```

Comment:
Without toString():
```
Assert.assertEquals("project matches entity", entity, project.toEntity());
```
java.lang.AssertionError: project matches entity expected: com.google.appengine.api.datastore.Entity<<Entity [Project(no-id-yet)]:
        creator = 2
        members = [1]
        name = Project 1
        description = Description of Project 1
        tasks = []
        admins = [3, 4]
>
> but was: com.google.appengine.api.datastore.Entity<<Entity [Project(no-id-yet)]:
        creator = 2
        members = [1]
        name = Project 1
        description = Description of Project 1
        tasks = []
        admins = [3, 4]

Reply:
Equality for two entities is true only if the key objects match:
https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/Entity.html#equals-java.lang.Object-

It sounds like it's not looking at the values itself, which is what you need.  Couple options you can do:  convert them as ProjectData and compare those (assuming you implemented the equals operation for your ProjectData).  Or create a reusable `CompareEntities(Entity e1, Entity e2)` method that extracts the properties using 'getProperties()' and compares them one by one